run_pretrained_openfold_metadata.py

```diff
@@ -0,0 +1,547 @@
+                    jackhmmer_binary_path=args.jackhmmer_binary_path,
+                    uniref90_database_path=args.uniref90_database_path,
+                    template_searcher=template_searcher,
+                    no_cpus=args.cpus,
+                )
+                embedding_generator = EmbeddingGenerator()
+                embedding_generator.run(tmp_fasta_path, alignment_dir)
+            else:
+                alignment_runner = data_pipeline.AlignmentRunner(
+                    jackhmmer_binary_path=args.jackhmmer_binary_path,
+                    hhblits_binary_path=args.hhblits_binary_path,
+                    uniref90_database_path=args.uniref90_database_path,
+                    mgnify_database_path=args.mgnify_database_path,
+                    bfd_database_path=args.bfd_database_path,
+                    uniref30_database_path=args.uniref30_database_path,
+                    uniclust30_database_path=args.uniclust30_database_path,
+                    uniprot_database_path=args.uniprot_database_path,
+                    template_searcher=template_searcher,
+                    use_small_bfd=args.bfd_database_path is None,
+                    no_cpus=args.cpus
+                )
+
+            alignment_runner.run(
+                tmp_fasta_path, local_alignment_dir
+            )
+        else:
+            logger.info(
+                f"Using precomputed alignments for {tag} at {alignment_dir}..."
+            )
+
+        # Remove temporary FASTA file
+        os.remove(tmp_fasta_path)
+
+
+def round_up_seqlen(seqlen):
+    return int(math.ceil(seqlen / TRACING_INTERVAL)) * TRACING_INTERVAL
+
+
+def get_representative_chain_ids(tags, fasta_path, metadata_cache):
+    """
+    This function maps the tags in the fasta file to the representative chain ids
+
+    Args:
+    tags: list of tags in the fasta file
+    fasta_path: path to the fasta file
+    metadata_cache: metadata cache that maps chain ids to pdb chain ids
+
+    Returns:
+    representative_chain_ids: list of representative chain ids corresponding to the tags in the fasta file
+    """
+    # Check if the pdb corresponding to the fasta file is found in the metadata cache
+    pdb_id = os.path.splitext(fasta_path)[0]
+    if pdb_id not in metadata_cache:
+        raise KeyError(f"PDB ID {pdb_id} not found in metadata cache")
+    # Check if the chains in the fasta file are found in the metadata cache
+    pdb_metadata = metadata_cache[pdb_id]
+    pdb_chains = set(pdb_metadata['chains'].keys())
+    is_subset = all([tag in pdb_chains for tag in tags])
+    if not is_subset:
+        raise KeyError(f"Chains from {fasta_path} not found in metadata cache")
+    # At this point we know that all chains in the fasta file are in the metadata cache
+    representative_chain_ids = [pdb_metadata['chains'][tag]['alignment_representative_id'] for tag in tags]
+    
+    return representative_chain_ids
```

Comment:
This function uses the metadata cashe to get representative chain ids corresponding to each chain in the fasta file